sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -1075,6 +1075,22 @@ def _add_argparse_args(cls, parser):
         ' directly, rather than starting up a job server.'
         ' Only applies when flink_master is set to a'
         ' cluster address.  Requires Python 3.6+.')
+    parser.add_argument(
+        '--parallelism',
+        default=-1,
+        type=int,
+        help='The degree of parallelism to be used when distributing '
+             'operations onto workers. If the parallelism is not set, the '
+             'configured Flink default is used, or 1 if none can be found.'
+    )
+    parser.add_argument(
+        '--execution_mode_for_batch',
```

Comment:
I'm not sure we should add these here because that's what we used to do and it was inconsistent and hard to maintain. What we want to do, is to not discard those options but still warn about them not being parsed in the Python SDK. This will allow us to still use them in the Runner code.

Reply:
I think we should continue to discard invalid options when the client is aware of the full set of options and can perform the validation - as is the case with portable runner and job server.

Reply:
> I think we should continue to discard invalid options when the client is aware of the full set of options and can perform the validation

Agreed, we would keep options only when we can't be sure (so just the uber jar job server for now).